packages/firestore/lite/src/api/snapshot.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as firestore from '../../index';
+
+import { Firestore } from './database';
+import { DocumentReference } from './reference';
+import { FieldPath } from './field_path';
+import { tryCast } from './util';
+import { DocumentKey } from '../../../src/model/document_key';
+import { Document } from '../../../src/model/document';
+import { UserDataWriter } from '../../../src/api/user_data_writer';
+import { FieldPath as InternalFieldPath } from '../../../src/model/path';
+import { fieldPathFromDotSeparatedString } from '../../../src/api/user_data_reader';
+
+export class DocumentSnapshot<T = firestore.DocumentData>
+  implements firestore.DocumentSnapshot<T> {
+  // Note: This class is stripped down version of the DocumentSnapshot in
+  // the legacy SDK. The changes are:
+  // - No support for SnapshotMetadata.
+  // - No support for SnapshotOptions.
+
+  constructor(
+    private _firestore: Firestore,
+    private _key: DocumentKey,
+    private _document: Document | null,
+    private _converter?: firestore.FirestoreDataConverter<T>
+  ) {}
+
+  get id(): string {
+    return this._key.path.lastSegment();
+  }
+
+  get ref(): firestore.DocumentReference<T> {
+    return new DocumentReference<T>(
+      this._firestore,
+      this._key,
+      this._converter
+    );
+  }
+
+  exists(): this is firestore.QueryDocumentSnapshot<T> {
+    return this._document !== null;
+  }
+
+  data(): T | undefined {
+    if (!this._document) {
+      return undefined;
+    } else {
+      // We only want to use the converter and create a new DocumentSnapshot
+      // if a converter has been provided.
+      if (this._converter) {
```

Comment:
Out of curiosity (i.e. no need to change), why create the nested block? Why not write it as `} else if (this._converter) {` ?